camel/toolkits/hybrid_browser_toolkit/ws_wrapper.py

```
@@ -892,20 +909,54 @@ async def mouse_control(
+            params = {'from_ref': from_ref, 'to_ref': to_ref}
+        elif all(v is not None for v in [from_x, from_y, to_x, to_y]):
+            params = {
+                'from_x': from_x,
+                'from_y': from_y,
+                'to_x': to_x,
+                'to_y': to_y,
+            }
+        else:
+            raise ValueError(
+                "Provide (from_ref, to_ref) or (from_x, from_y, to_x, to_y)"
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)